### ISSUE TITLE
Twitch and Microsoft TOTP invalid after a while

### ISSUE THREAD

**Issue by SuperSandro2000** [andOTP/andOTP#545]
#### General information

* **App version:** 0.7.0-play
* **App source:** Google Play
* **Android Version:** 9
* **Custom ROM:** ALEXNDR

#### Expected result
**What is expected?** 

That everything just works and I do not need to reimport entries after a while.

**What does happen instead?**

The Twitch Authy token got invalidated after a while and I needed to re-import this. Today I noticed that the Microsoft token was also invalid. I have no idea why and how. I can only imagine that either the algorithm is faulty and gets more incorrect after a while or backup restores/titanium backup somehow mess it up. I use the app more than daily with other services and they just work fine and never have such problems.

#### Logcat
There is no relevant information or crashes.

#### Steps to reproduce
 
 * Import Authy Twitch, Microsoft
 * Wait?
 * In X amount of time it is broken

**Comment by beerisgood** [comment]
Is your device time correct?

**Comment by SuperSandro2000** [comment]
I looked closer on the Microsoft token and in the native app it has 8 digits and the imported one has 6. No wonder that this is not working. I am catching up with https://github.com/puddly/android-otp-extractor and try to get this solved.

And my Device time is correct. I never set it by hand. It was always set by mobile connection which is correct and all other services 50+ work fine. It was just once that Twitch desynced and Microsoft was totally wrong.

**Comment by sanawu** [comment]
> I looked closer on the Microsoft token and in the native app it has 8 digits and the imported one has 6. No wonder that this is not working. I am catching up with https://github.com/puddly/android-otp-extractor and try to get this solved.
> 
> And my Device time is correct. I never set it by hand. It was always set by mobile connection which is correct and all other services 50+ work fine. It was just once that Twitch desynced and Microsoft was totally wrong in the beginning

This app doesn't support 8 digit TOTP? Bwahahahahaha

**Comment by flocke** [comment]
andOTP supports 8-digit TOTP tokens, but the service has to provision them correctly (by including the digits=8 parameter in the QR code).

**Comment by flocke** [comment]
Is this problem still happening or can we close this issue?

**Comment by Yrlish** [comment]
I experienced something similar. I configured my Twitch 2FA with andOTP, it worked. Then I applied to remove my Authy account completely. This resulted with my TOTP in andOTP stopped working and apparently the only backup Twitch have is SMS (through Authy)...

**Comment by flocke** [comment]
I'll have to look into this, maybe Twitch changed something on their end.

But we need to remember that Twitch does not officially support plain TOTP. They might have changed something to break 3rd party apps, which might mean we won't be able to get it working again.

**Comment by Yrlish** [comment]
No, Twitch has added support to TOTP as an alternative to the Authy app, but it still seems to be dependent on Authy somehow. Twitch's entire 2FA implementation is extremely dumb.

**Comment by flocke** [comment]
Seems like I have to dig out my old Twitch account and do some digging.
